src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResponseResource.java

```diff
@@ -62,26 +59,66 @@ public ComplaintResponseResource(ComplaintResponseRepository complaintResponseRe
+     *
+     * @param complaintId - id of the complaint to lock again
+     * @return the ResponseEntity with status 201 (Created) and with body the empty complaint response
+     */
+    @PostMapping("/complaint-responses/complaint/{complaintId}/refresh-lock")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ComplaintResponse> refreshLockOnComplaint(@PathVariable long complaintId) {
+        log.debug("REST request to refresh empty complaint response for complaint with id: {}", complaintId);
+        ComplaintResponse savedComplaintResponse = complaintResponseService.refreshEmptyComplaintResponse(complaintId);
         // always remove the student from the complaint as we don't need it in the corresponding client use case
-        complaintResponse.getComplaint().filterSensitiveInformation();
+        savedComplaintResponse.getComplaint().filterSensitiveInformation();
+        return new ResponseEntity<>(savedComplaintResponse, HttpStatus.CREATED);
+    }
 
-        return ResponseEntity.created(new URI("/api/complaint-responses/" + savedComplaintResponse.getId()))
-                .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, entityName, savedComplaintResponse.getId().toString())).body(savedComplaintResponse);
+    /**
+     * DELETE /complaint-responses/complaint/:complaintId/remove-lock: removes the lock on a complaint by removing the empty complaint response
+     *
+     * @param complaintId - id of the complaint to remove the lock for
+     * @return the ResponseEntity with status 200 (Ok)
+     */
+    @DeleteMapping("/complaint-responses/complaint/{complaintId}/remove-lock")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Void> removeLockFromComplaint(@PathVariable long complaintId) {
+        log.debug("REST request to remove the lock on the complaint with the id: {}", complaintId);
+        complaintResponseService.removeEmptyComplaintResponse(complaintId);
+        return ResponseEntity.ok().build();
+    }
+
+    /**
+     * PUT /complaint-responses/complaint/:complaintId/resolve: resolve a complaint by updating the complaint and the associated empty complaint response
+     *
+     * @param complaintId - id of the complaint to resolve
+     * @param complaintResponse the complaint response used for resolving the complaint
+     * @return the ResponseEntity with status 200 (Ok) and with body the complaint response used for resolving the complaint
+     */
+    @PutMapping("/complaint-responses/complaint/{complaintId}/resolve")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ComplaintResponse> resolveComplaint(@RequestBody ComplaintResponse complaintResponse, @PathVariable long complaintId) {
+        log.debug("REST request to resolve the complaint with id: {}", complaintId);
+        ComplaintResponse updatedComplaintResponse = complaintResponseService.resolveComplaint(complaintResponse);
```

Comment:
seems that permission checks are missing here

Reply:
Done in the service. See:
private boolean isUserAuthorizedToRespondToComplaint(StudentParticipation participation, Complaint